src/plugins/data/server/search/strategies/sql_search/response_utils.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { SqlQueryResponse } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { SqlSearchStrategyResponse } from '../../../../common';
+
+/**
+ * Get the Kibana representation of an async search response
+ */
+export function toAsyncKibanaSearchResponse(
+  response: SqlQueryResponse,
+  warning?: string
+): SqlSearchStrategyResponse {
+  return {
+    id: response.id,
+    rawResponse: response,
+    isPartial: response.is_partial,
+    isRunning: response.is_running,
+    ...(warning ? { warning } : {}),
+  };
+}
```

Comment:
Same feedback here as above... can we use the method from `ese_search/response_utils` here?

Reply:
They are quite different. I think combining them would bring more complexity than benefit